### ISSUE TITLE
bug(web): underlying key display flags not being respected

### ISSUE THREAD

**Issue by MakaraSok** [keymanapp/keyman#3950]
**Describe the bug**
Web emulator shows a keyboard layout without underlying characters, while the Android emulator shows all underlying characters.

![image](https://user-images.githubusercontent.com/28331388/100189061-1d2dd780-2f1e-11eb-8262-05f6c0fb1ccc.png)

**To Reproduce**
This behavior was encountered when working on this project: 
[bj_naskapi_classic.zip](https://github.com/keymanapp/keyman/files/5594687/bj_naskapi_classic.zip)

**Expected behavior**
The touch layouts should be shown consistently on both web and Android emulators.

---

**Android emulator:**
 - Device: Pixel 2
 - OS: Android 10
 - Keyman version 14.0.188
 - Target application: [Keyman for Android]

**KeymanWeb:**
 - OS: Windows 10 Pro 1909
 - Browser: Chrome 87.0.4280.66 (Official Build) (64-bit)
 - KeymanWeb Version: [?]
 - URL: N/A
 
**Keyman Developer:**
 - OS: Windows 10 Pro 1909
 - Keyman Developer version: 13.0.115
 
**Additional context**
The Desktop OSK in .KMN has `Display underlying layout characters` checked.

**Comment by mcdurdin** [comment]
Is the `displayUnderlying` flag set for both tablet and phone layouts?

**Comment by MakaraSok** [comment]
> Is the `displayUnderlying` flag set for both tablet and phone layouts?

No. There is no phone layout. The checkbox for `Display underlying` is not checked either.

![image](https://user-images.githubusercontent.com/28331388/100189879-c45f3e80-2f1f-11eb-90cb-e81ff84a4cf5.png)

It is also observed that when the Desktop OSK in .KMN has `Display underlying layout characters` unchecked, the underlying characters on keycaps on Android emulator

![image](https://user-images.githubusercontent.com/28331388/100189836-ae517e00-2f1f-11eb-8161-60993db85532.png)

**Comment by darcywong00** [comment]
> No. There is no phone layout. The checkbox for Display underlying is not checked either.

However, the compiled touch keyboard bj_naskapi_classic.js has display "underling characters" (KDU) set

```
})(this.KV.KLS);
  this.KDU=1;
  this.KH='';
  this.KM=0;
  this.KBVER="2.1";
```

Maybe Developer is using the kvks flag and applying to touch keyboard?

**Comment by jahorton** [comment]
Just investigated this on the current `beta` branch.  I don't see any inconsistency in "display underlying" flag use.

<img width="391" alt="Screen Shot 2021-01-14 at 12 28 06 PM" src="https://user-images.githubusercontent.com/25213402/104548881-4d633e00-5664-11eb-9e68-995644f3979f.png">
<img width="327" alt="Screen Shot 2021-01-14 at 12 28 57 PM" src="https://user-images.githubusercontent.com/25213402/104548888-518f5b80-5664-11eb-9da8-b478f7968956.png">
<img width="747" alt="Screen Shot 2021-01-14 at 12 29 34 PM" src="https://user-images.githubusercontent.com/25213402/104548890-53f1b580-5664-11eb-8086-36a4aca167b8.png">

Sounds like the main bug here is the Developer part - where "display underlying" is intended for desktop platform use only, yet Web's equivalent flag is also being set.

**Comment by MakaraSok** [comment]
Retested with 14.0.221 using the zip folder above. When `Display underlying` for Touch Layout is not checked in Developer, both Web and Mobile version still show the underlying characters.

![image](https://user-images.githubusercontent.com/28331388/104561327-d173f080-5679-11eb-9ce7-613260cef8c5.png)

**Comment by mcdurdin** [comment]
> However, the compiled touch keyboard bj_naskapi_classic.js has "display underling characters" (KDU) set

## Keyboard source 

Just to be clear, there are two 'display underlying' flags in the source keyboard:

* On Screen Keyboard 'display underlying' flag is **true**. This applies to desktop on screen keyboards. In the .kvks file:

```xml
    <flags>
      <displayunderlying/>
    </flags>
```

* Touch Keyboard, tablet platform 'display underlying' flag is **false**. This applies to touch layouts (both phone and tablet as there is no separate phone layout). In the .keyman-touch-layout file:

```js
{
  "tablet": {
    ...
    "displayUnderlying": false
  }
}
```

## Compiled result

The keyboard shows the following, compiled, which is correct as far as I can tell:

```js
this.KDU=1;
````

`this.KDU` is the flag for the desktop OSK, and should not be used for the touch layout.

```js
  this.KVKL={
  "tablet": {
    "font": "Tahoma",
    "displayUnderlying": false,
```

`this.KVKL.<platform>.displayUnderlying` is the flag that should be used for touch devices.

Looking at the history, this was implemented in #2353 but that code has disappeared since then and it looks like a refactor in KeymanWeb is treating all platforms identically (`displaysUnderlyingKeys()` in keyboard.ts references only `KDU`):

https://github.com/keymanapp/keyman/blob/f7a21e6d848444111832fb370bca31d56bf7a2b9/web/source/osk/visualKeyboard.ts#L620-L624

versus (14.0 beta):

https://github.com/keymanapp/keyman/blob/6bb00124ee5129bed828e63e167ce7e3b35087a6/web/source/osk/visualKeyboard.ts#L692-L693

**Comment by jahorton** [comment]
Hmm... that does leave one last question.  Suppose the following scenario:

A keyboard that:
- does not specify a touch layout, relying upon KMW defaults
- does specify the `KDU` property
- is being used on a touch device.

Since the resulting layout is "desktop-based", and there is no layout pre-defined for that form factor, should this scenario utilize the flag or not?
